test/gurka/test_avoids.cc

```diff
@@ -214,28 +186,50 @@ TEST_P(AvoidTest, TestAvoid2Polygons) {
   };
 }
 
-TEST_P(AvoidTest, TestAvoidLocation) {
-  auto node_a = avoid_map.nodes.at("A");
-  auto node_b = avoid_map.nodes.at("B");
-  auto dx = std::abs(node_b.lng() - node_a.lng());
+TEST_P(AvoidTest, TestAvoidShortcuts) {
+  valhalla::Options options;
+  options.set_costing(valhalla::Costing::truck);
+  auto* co = options.add_costing_options();
+  co->set_costing(valhalla::Costing::truck);
+
+  // create the polygon intersecting a shortcut
+  auto* rings = options.mutable_avoid_polygons();
+  auto* ring = rings->Add();
+  for (const auto coord :
+       {avoid_map.nodes["p"], avoid_map.nodes["q"], avoid_map.nodes["r"], avoid_map.nodes["s"]}) {
+    auto* ll = ring->add_coords();
+    ll->set_lat(coord.lat());
+    ll->set_lng(coord.lng());
+  }
+
+  const auto costing = valhalla::sif::CostFactory{}.Create(*co);
+  GraphReader reader(avoid_map.config.get_child("mjolnir"));
 
-  auto lon = node_a.lng() + 0.5 * dx;
+  // should return the shortcut edge ID as well
+  const auto avoid_edges = vl::edges_in_rings(*rings, reader, costing, 10000);
 
-  // avoid the "High" road ;)
-  //      A---x--B
-  //      |      |
-  //      |      |
-  //      |      |
-  //      D------E
-  std::vector<vm::PointLL> points{{lon, node_a.lat()}};
+  size_t found_shortcuts = 0;
+  for (const auto& edge_id : avoid_edges) {
+    if (reader.GetGraphTile(edge_id)->directededge(edge_id)) {
+      found_shortcuts++;
+      break;
+    }
+  }
+
+  ASSERT_EQ(found_shortcuts, 1);
```

Comment:
maybe i forgot a detail here, dont we add the edge and its opposing edge? why isnt the count 2?

Reply:
Ah thanks I wondered the same thing! It's not the avoid polygon algo, the graph has simply 1 shortcut (I printed it out before).. I figured it might be bcs the other direction is pretty pointless for a shortcut and the graphbuilder is smart enough to omit it (i.e. going from E -> C one could never use that shortcut since search locations will never project on it?!)?

Reply:
that cant be possible! every edge has to have an opposing edge otherwise the graph is broken. the detail is that we only ever put one of the twin edges into the bins. we should, for both loki::search avoid edges and edges_in_ring avoided edges, always also return the opposing edge ids because if we dont then the algorithm will randomly be allowed to use one of the side of the intersecting roads and not the other.

Reply:
it's the same as here no?

https://github.com/valhalla/valhalla/blob/d15ae9f69968a2c2008c1b3d0cddd15783f4bef0/test/gurka/test_shortcut.cc#L260-L291

Reply:
and it seems we correctly do! https://github.com/valhalla/valhalla/blob/master/src/loki/polygon_search.cc#L132-L133

Reply:
what am I missing here then?

Reply:
oh sorry i didnt notice you were looking at the bins in this test, i got confused about which test it was... apologies!

Reply:
uff.. copy/paste fail.. there were never any shortcuts after I re-arranged the map (plus some other stupid things..).. added another node to make the shortcut.